Adds updated, fixed versions of the Sawustation SNPCs.

sawu-tg opened the pull request:
They mostly function but have a few minor problems, but are otherwise working.

Includes a module system to add your own features and abilities to them. Mostly bug and lag free.

Known issues include:
*They will path through walls sometimes (resulting in standing and staring at it).
*They occasionally forget that they are holding large objects and try to ram them in pockets instead of backpack (maybe fixed?)
*Have been known to hold grudges (..Feature)
*Uses items with item.attack() so any special snowflakes can not be used by them.
*They have a tendancy to be loud, noisy and rude (but that's just an average player, right?) (..Feature)

Summary of the features in one giant sentence (also down below):

Currently, they can imitate players at a base form. They're created to copy a player, assigned a job (to which the department is taken into consideration for walking etc), They can wield weapons, fire guns, beat people with things and theoretically use any item that does things on item.attack().

The base form of the NPC will wander around, pick up items (%chance according to var), wield them, switch out clothes, retaliate to attacks and act like a regular (bad) player in general.

Additional modules (loaded by adding them to the module, just basic procs), which are included allow it to handcuff people, spew randomised bullshit, does advanced fighting (by default, only disarms/whacks in response), and keeps a list of all internal doors for special procs.

They have (currently) three pre-designed sets of "personalities", which are just sets of the personality variables. "Friendly" which attempts to stay away from combat, "Mean" which is more prone to combat, and "Greytide" which is full on shitler (and spawns with no items)

The NPCs are designed for admin events, adding to other things (away missions? events? holidays? maybe replace some of the more advanced simple_mobs), and exist mainly as a base to be improved upon and added to, especially with the ease of the modules.

sawu-tg committed:
Adds updated, fixed versions of the Sawustation SNPCs, they mostly function but have a few minor problems, but are otherwise working.

MMMiracles commented:
Fun fact with this: if the job selector for them chooses captain and there isn't already one on-station, it'll announce their arrival.

sawu-tg commented:
@MMMiracles Double-Fun fact: If they job selector chooses them as Captain and one already exists, it doesn't give a crap and makes them Captain anyway, resulting in multiple Captains :hocho:. I call it a feature.

JJRcop commented on code/modules/mob/interactive.dm:
```diff
@@ -0,0 +1,643 @@
+		var/turf/T = get_step(src,dir)
+		if(T)
+			for(var/obj/machinery/door/D in T.contents)
+				if(D.check_access(MYID) && !istype(D,/obj/machinery/door/poddoor) && D.layer == 3.1)
```
Instead of this, why not simply check for density, which is what the door open() proc actually does.

sawu-tg committed:
Quickfixes for SNPCs
*Adds a variable for the chattyness of NPCs
*Removes unneeded variable
*Fixes NPCs using weapons/items after dropped
*Fixes some combat problems, makes them more robust in general

CorruptComputer commented on code/modules/mob/interactive.dm:
```diff
@@ -0,0 +1,660 @@
+	var/list/nouns_generic = list("you", "you there", "guy", "person", "dude", "shitbin", "officer")
+	var/list/nouns_objects = list("thing","tool","whatever","potential murder weapon")
+	var/list/nouns_body = list("ass","head","face","arms","fingers","toes","chest","eyes","nails","hoo-hah")
+	var/list/adjective_insult = list("ugly","repulsive","dumb","asian","childish","captain-like","monkey-like","whoreish")
```
How is asian an insult?

sawu-tg replied:
It's not, it's an adjective (in this sense). I threw it in there because it creates some mildly amusing generated sentances: 
![lolwoops](https://cloud.githubusercontent.com/assets/3829126/5601582/2eb555f4-935d-11e4-8312-cb88725b2859.PNG)
If it's too offensive/racist, I can remove it.

CorruptComputer replied:
I'm not offended by it, but I just didn't see how it was an insult, but seeing it makes it 200% better. :+1:

Tenebrosity replied:
oh my

sawu-tg committed:
Change/fix to door checks.
*Uses density now instead of layer.

sawu-tg committed:
Changes to speech, comments
*Added a couple of extra comments
*Added a bunch of extra words to the NPC's vocab, along with a suggestion.

Nienhaus commented:
:+1: x :100:

sawu-tg commented:
P.S. Eat your heart out  #6730

paprka commented:
What do these do again?

ike709 commented:
they took 'er jerbs!

Miauw62 commented:
As an early playtester, I can confirm that this is 100% hilarious, merge he
posthaste.
Op 3-jan.-2015 06:30 schreef "ike709" <notifications@github.com>:

> they took 'er jerbs!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/6762#issuecomment-68584189>
> .
>

sawu-tg commented:
@paprka Currently, they can imitate players at a base form. They're created to copy a player, assigned a job (to which the department is taken into consideration for walking etc), They can wield weapons, fire guns, beat people with things and theoretically use any item that does things on item.attack().

The base form of the NPC will wander around, pick up items (%chance according to var), wield them, switch out clothes, retaliate to attacks and act like a regular (bad) player in general.

Additional modules (loaded by adding them to the module, just basic procs), which are included allow it to handcuff people, spew randomised bullshit, does advanced fighting (by default, only disarms/whacks in response), and keeps a list of all internal doors for special procs.

They have (currently) three pre-designed sets of "personalities", which are just sets of the personality variables. "Friendly" which attempts to stay away from combat, "Mean" which is more prone to combat, and "Greytide" which is full on shitler (and spawns with no items)

The NPCs are designed for admin events, adding to other things (away missions? events? holidays? maybe replace some of the more advanced simple_mobs), and exist mainly as a base to be improved upon and added to, especially with the ease of the modules.

Iamgoofball commented:
oh god i remember these

tried to code them crafting spears

they impaled themselves on glass shards


10/10 merghe post-haste

Iamgoofball commented on code/modules/mob/interactive.dm:
```diff
@@ -0,0 +1,661 @@
+/mob/living/carbon/human/interactive/proc/job2area(var/target)
+	var/datum/job/T = target
+	if(T.title == "Assistant")
+		return /area/hallway/primary
```
>not returning permabrig

Iamgoofball commented on code/modules/mob/interactive.dm:
```diff
@@ -0,0 +1,661 @@
+	var/list/verbs_use = list("touch","hold","poke","punch","smash","assault","wrench","tickle","modify","whack","beat","tackle","fiddle","destroy")
+	var/list/verbs_touch = list("stroke","pat","lick","finger","mess with","absorb","sneeze on","fist","feel up","inject","rip")
+	var/list/verbs_move = list("run","jog","leg it","slide","cheese it","walk","stroll","fly","scoot","yakkety","speed","fuck off","escape")
+	var/list/nouns_insult = list("faggot","shitcurity","noob","comdom","greyturd","tator","lingbin","bitch","whorelord","cluwne","shitbin","turdcurity","fuckblanket","asswhistle","cumdumpster")
```
i object to the slurs
everything else is fine

ike709 replied:
the slurs are great

sawu-tg replied:
@Iamgoofball Which slurs do you object to? I used language pretty tame for /tg/ players imo.

Iamgoofball replied:
faggot is the only one I really don't like

everything else seems average for a tg player

sawu-tg replied:
@Iamgoofball If more people oppose it, I will remove it, but as it goes, I see every second player use the word these days. I personally take no offence in it.

optimumtact replied:
Can you please pull them from a text file? See the AI ion law list for an example, also this PR

https://github.com/tgstation/-tg-station/pull/5884

sawu-tg replied:
Can do.

sawu-tg committed:
Changes to the Chatter module.
*NPCs now use pick_list to draw their speech from strings/npc_chatter.txt

Dorsisdwarf commented:
There should be 1-2 of these in the unused areas of the station at all times. And a clown in the permabrig.

Nienhaus commented:
Oh man, a chance for a clown to spawn in the permabrig would be great.

sawu-tg commented:
@Dorsisdwarf What do you mean unused areas? Do you mean like maintenance? or Do you mean check crew roles and insert them in the locations of empty crewmembers?

ike709 commented:
I think that a couple assistants should be lurking in maint.

Ergovisavi commented:
Parker Knapenberger says, "Well cunt, this is a pretty situation."
Parker Knapenberger says, "Oh bitch, that bitch was one hell of an dumb groin."
Parker Knapenberger exclaims, "Hey you, why dont you go ass cunt bitch bitch cunt , you bitch!"
Parker Knapenberger says, "Well bitch, this is a stupid situation."
Parker Knapenberger says, "Well modify my head, this greyturd is pretty vomit-inducer."
Parker Knapenberger says, "I want to poke that whorelord when I find them."

My only complaints:
>1)Please change them so that they're not so blatantly foul mouthed / state sexually suggestive things
>2)Maybe tone down their speed from SANIC
>3)Change the examine description to not have the comatose message
>4) When fighting they occasionally do long periods of nothing, especially when fighting simple mobs. Any idea what's up with this?

Miauw62 commented:
it was inevitable

sawu-tg committed:
Ergo's Changes
*Examine shows NPCs a unique message
*NPCs should no longer stand and stare at people in combat.
*Toned down the amount of rudeness in their speech.

sawu-tg committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into interactive

sawu-tg committed:
Quickfix to conform with the rewrite.

Ergovisavi commented:
Travis failing is just travis being travis, tested this locally and everything appears to function.

JJRcop commented:
Well, travis is failing because asteroid and ministation are erroring.

Iamgoofball commented:
amazing 10/10 i love it

sawu-tg commented:
People wishing to write modules, [Click here](https://dl.dropboxusercontent.com/u/1513925/ss13/module_writing_for_snpcs.txt)

Miauw62 commented:
>it even has documentation

Is this merge-ready? I just love it.
Op 7-jan.-2015 04:15 schreef "Sawu" <notifications@github.com>:

> People wishing to write modules, Click here
> <https://dl.dropboxusercontent.com/u/1513925/ss13/module_writing_for_snpcs.txt>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/6762#issuecomment-68973622>
> .
>

RemieRichards commented:
@Miauw62 
>is this merge-ready?

it's been merged... you fool...

Miauw62 commented:
In my defense, I didn't see
Op 7-jan.-2015 12:43 schreef "Remie Richards" <notifications@github.com>:

> @Miauw62 <https://github.com/Miauw62>
>
> is this merge-ready?
>
> it's been merged... you fool...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/6762#issuecomment-69011492>
> .
>

JJRcop commented:
Maybe in the future someone could convert our simple_animal antags to this
